pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java

```diff
@@ -111,6 +113,15 @@ private static BaseFilterOperator constructPhysicalOperator(FilterQueryTree filt
       // Leaf filter operator
       Predicate predicate = Predicate.newPredicate(filterQueryTree);
 
+      // Check for null predicate
+      Predicate.Type type = predicate.getType();
```

Comment:
move this code to FilterOperatorUtils.getLeafOperator? we can have an IsPredicateEvaluator that return empty list for matchings and non matching dictIds

Reply:
Yeah - couldn't figure out how to do this in predicate evaluator. Leaving it as is for now.